assets/js/lib/model/checks.js

```diff
@@ -0,0 +1,28 @@
+import { pipe, some, castArray, filter } from 'lodash/fp';
+import { TARGET_CLUSTER } from '.';
+
+const hasMetadata = ({ metadata }) => !!metadata;
+const hasClusterTypeMetadata = ({ metadata }) =>
+  hasMetadata({ metadata }) && !!metadata.cluster_type;
+
+const isCheckForTarget =
+  (targetType) =>
+  ({ metadata }) =>
+    !hasMetadata({ metadata }) || metadata.target_type === targetType;
+
+export const hasChecksForTarget = (catalog, targetType) =>
+  some(isCheckForTarget(targetType))(catalog);
+
+export const hasChecksForClusterType = (catalog, clusterType) =>
+  pipe(
+    filter(isCheckForTarget(TARGET_CLUSTER)),
+    some(
+      ({ metadata }) =>
+        !hasMetadata({ metadata }) ||
+        !hasClusterTypeMetadata({ metadata }) ||
+        pipe(
```

Comment:
Isn't this creating an array of size 1?

Reply:
you mean `castArray`?

If you're referring to it, it creates an array of size 1 if cluster_type is let's say a string like `ascs_ers`, but it covers cases where we have:
```yaml
metadata:
  target_type: cluster
  cluster_type:
    - hana_scale_up
    - hana_scale_out
```
maybe not frequent, but the metadata in the check work like this :smile:

Reply:
Oh okay, so the type of `clusterType` is `String | [String]`?

Reply:
potentially yes